bee-protocol/src/workers/message/hash_cache.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::{HashSet, VecDeque},
+    hash::{BuildHasherDefault, Hasher},
+};
+
+use twox_hash::XxHash64;
+
+struct CustomHasher {
+    result: u64,
+}
+
+impl CustomHasher {
+    fn finish(&self) -> u64 {
+        self.result
+    }
+    fn write(&mut self, i: u64) {
+        self.result = i;
+    }
+}
+
+impl Default for CustomHasher {
+    fn default() -> Self {
+        Self {
+            result: 17_241_709_254_077_376_921,
+        }
+    }
+}
+
+impl Hasher for CustomHasher {
+    fn finish(&self) -> u64 {
+        CustomHasher::finish(self)
+    }
+    fn write(&mut self, bytes: &[u8]) {
+        let (int_bytes, _rest) = bytes.split_at(std::mem::size_of::<u64>());
```

Comment:
why not `bytes[..std::mem::size_of::<u64>()]`? 

Also,  both options panic if there aren't enough bytes, why do we know that there are `8` bytes all the time?